src/meta/src/hummock/hummock_manager.rs

```diff
@@ -591,12 +591,19 @@ where
                     .collect::<HashSet<u32>>();
                 compact_task.vnode_mappings.reserve_exact(table_ids.len());
                 for table_id in table_ids {
-                    let vnode_mapping = self
-                        .cluster_manager
+                    if let Some(vnode_mapping) = self
+                        .env
+                        .hash_mapping_manager()
                         .get_table_hash_mapping(&table_id)
```

Comment:
Current logic is to set vnode mapping into compaction task only when vnode mapping of the table exists. There's no check on whether the compaction is hash or range compaction. Do we need to add some checks to ensure that hash compaction always has non-None vnode mappings? cc. @hzxa21

Reply:
> Since TopN is always singleton, it could be allowed not to have a vnode mapping. Compactor will omit the part of hash compaction according to the fact that vnode mapping is None, even if it is a stateful operator and should do hash compaction.

If we adopt this, then it seems no need to have such check.